### PR TITLE
Frontend: improved the button component

### LINKED ISSUE #120: Improve the Button library component.
# Description
The button component works fine, but exposes too many options as props, when in reality, we might never need more than a small set of configurable options.

We should make the button component a bit more user friendly by providing some more default options.

```
type ButtonProps = {
    text: string;
    textColor?: string;
    backgroundColor?: string;
    fontSize?: string;
    rounded?: string;
    buttonSize?: string;
    type?: 'submit' | 'reset' | 'button';
    borderColor?: string;
    hoverColor?: string;
    hoverScale?: boolean;
    px?: string;
    py?: string;
    className?: string;
    onClick?: () => void;
};
```

- The button type should be sufficient to determine the `borderColor`, `hoverColor` etc. Let's go with the `accept` and `reject` types for now, which have a bunch of preconfigured appearance - accept is green, reject is red etc. If the type is an empty string, we then consider the values of the other props for styling. The hoverColor can be replaced by a brightness filter.
- We can similarly introduce a size prop from which we can infer some predetermined values for `rounded`, `px`, `py`, `fontSize` etc.
- We should accept the button content as children, not as props.
- We don't need to remove the preexisting options - just add some defaults over them.

# Notes
- Post some sample button screenshots using the implemented defaults in the PR description.

### PR DESCRIPTION
**Pull request by sksmagr23**
## Description
The button component is modified to make it more user friendly ,and added default options for type and size props like reject, accept, rounded font size, etc. also brightness filter added in hover colour as default, and button content accepted as children rather than props.
The corresponding button component used everywhere in site is modified as per modified button component.

fixes #120

## How to Test
- Run npm run dev in frontend directory

## Related Issues
- #120 

## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have assigned reviewers to this pull request.

## Screenshots 
Screenshot from 2024-06-18 11-52-15

Screenshot from 2024-06-18 11-52-29

### CODE REVIEW

File: frontend/src/Navbar.tsx [hunk 1]
```diff
@@ -83,51 +85,62 @@ const Navbar: React.FC = () => {
...
                         ) : (
                             <>
                                 <Button
-                                    text="Login"
-                                    onClick={() => navigate('/login')}
-                                    className="border-4 mb-2 mt-20"
+                                    variant="accept"
+                                    size="medium"
                                     fontSize="text-2xl"
                                     buttonSize="w-[170px] h-12"
```
Comment (kuv2707) [inline frontend/src/Navbar.tsx:105]: Can't we infer the buttonSize from the size prop?
we can set the width to fit-content and choose the height based on the font size

File: frontend/src/pages/Error.tsx [hunk 2]
```diff
@@ -24,12 +24,14 @@ function Error() {
...
-                        text="Return back to homepage"
+                        variant="accept"
+                        size="medium"
                         buttonSize="w-80 h-12"
-                        className="border-4"
                         rounded="rounded-full"
                         onClick={() => navigate('/')}
-                    />
+                    >
+                        Return back to homepage
```
Comment (kuv2707) [inline frontend/src/pages/Error.tsx:33]: Change the text to "Return to homepage"

File: frontend/src/pages/SignUp.tsx [hunk 3]
```diff
@@ -95,14 +95,16 @@ const SignUp: React.FC = () => {
...
-                                            py="0"
-                                            buttonSize="w-32 h-10"
-                                            className="border-4 rounded-full"
-                                            fontSize="text-2xl"
+                                            variant="accept"
+                                            size="medium"
+                                            buttonSize="w-36 h-10"
                                             type="submit"
-                                        />
+                                            rounded="rounded-full"
```
Comment (kuv2707) [inline frontend/src/pages/SignUp.tsx:102]: We can set `rounded` based on the size.

File: frontend/src/Navbar.tsx (hunk 1 above)
Comment (sksmagr23) [reply frontend/src/Navbar.tsx:105]: Bhaiya actually there are many size of buttons used so default one is not used much

File: frontend/src/pages/Login.tsx [hunk 4]
```diff
@@ -75,14 +75,16 @@ const Login: React.FC = () => {
...
-                                            buttonSize="w-32 h-10"
-                                            className="border-4 rounded-full"
-                                            fontSize="text-2xl"
+                                            variant="accept"
+                                            size="medium"
+                                            buttonSize="w-36 h-10"
                                             type="submit"
-                                        />
+                                            rounded="rounded-full"
+                                            py="py-0"
```
Comment (kuv2707) [inline frontend/src/pages/Login.tsx:83]: We should be able to delete this prop

**Review by kuv2707 — COMMENTED** [review]
This looks great!
Left some comments. 
Overall, we'd want the `size` prop to handle more default values for other properties.

All/Most of the the buttons currently present in the website should be able to be described by the `size` and the `color` props.

Also it should be assured that explicitly setting a prop value should override the default value provided to it by the `size` prop.

File: frontend/src/Navbar.tsx (hunk 1 above)
Comment (sksmagr23) [reply frontend/src/Navbar.tsx:105]: But through I have tried to add buttonsize in default size properly , but somewhere it is added separately as per width needs

**Comment by sksmagr23** [comment]
Done the reviewed changes

**Comment by kuv2707** [comment]
@sksmagr23 It is a good practice to respond to the feedback if you have addressed it. It is the maintainer's job to then mark the threads resolved, if the problem is actually addressed.

**Comment by kuv2707** [comment]
Merged, thanks @sksmagr23 !

This must be sufficient to get the initial prototype working

**Pull request merged**